system/jlib/jio.cpp

```diff
@@ -1387,42 +1387,23 @@ class CStreamLineReader : public CSimpleInterfaceOf<IStreamLineReader>
                 {
                     ++currentPtr;
                     out.append(currentPtr-startPtr-(preserveEols?0:1), startPtr);
-                    if (currentPtr == endPtr)
-                        return !refill();
                     return false;
                 }
                 case '\r':
                 {
                     ++currentPtr;
-                    // check for \n
-                    if (currentPtr < endPtr)
-                    {
-                        if ('\n' == *currentPtr) // i.e. \r\n
-                        {
-                            ++currentPtr;
-                            out.append(currentPtr-startPtr-(preserveEols?0:2), startPtr);
-                        }
-                        else // i.e. \r only
-                            out.append(currentPtr-startPtr-(preserveEols?0:1), startPtr);
-                        if (currentPtr == endPtr)
-                            return !refill();
-                        return false;
-                    }
-                    else // must output what we have and read 1 more byte to check for \n
-                    {
-                        out.append(currentPtr-startPtr-(preserveEols?0:1), startPtr);
-                        if (!refill())
-                            return true;
-                        else if ('\n' != *currentPtr)
-                            return false;
-                        ++currentPtr;
+                    out.append(currentPtr-startPtr-(preserveEols?0:1), startPtr);
```

Comment:
this change introduces a subtle change - lines that terminate in \r\n will write all but 1 byte, and then append the \n.  That *might* cause the target buffer to reallocate.  I doubt it is significant though.

Reply:
There doesn't seem to be any need to preserve line terminators in any of the current code that uses CStreamLineReader.  We could set all of them to preserveEols==false? (I can also add a comment to state the point about the in the code about this so someone can modify if the need arises)